src/view/caselists/SnoozedCaseList.tsx

```diff
@@ -99,7 +109,7 @@ const UnconnectedSnoozedCaseList = (props: Props) => {
     setCases(snoozedCases);
   };
 
-  return (
+  return caseType === "snoozed" ? (
```

Comment:
To me it feels like the loading state should be checked with in the `CaseList` component.

I think we should add a loading cases flag that is set when the case list API call is made and then restructure how we store cases. We could clear the `caselist` on `caseType` change. or make a nested caselist object:
```
interface Caselist {
     active: Case[];
     snoozed: Case[];
}
```

Then we can display `<p>Loading</p>` when `isCaseListLoading === true && caselist.length === 0`

Reply:
I agree the current way of handling loading, especially when switching between routes, is clunky. That being said, I the nested caselist object seems like it will add complexity without enough benefit. Maintaining two lists means we need to make sure we're performing whatever operations on the correct list, but with one list we always know we're dealing with the correct one.

Reply:
I pushed a commit that removes the ` return caseType === "snoozed" ? ... : <p>Loading</p>` logic since it was clunky. If you spin the app up locally, you'll see the app essentially functions fine, but when you switch from the "Active Cases" to the "Snoozed Cases" tab, you'll get a bunch of console errors. This is happening because the case list clearing/refreshing is being triggered in the `ActiveCaseList` and `SnoozedCaseList` `useEffect` hook, resulting in a render _before_ the effect is actually run. (It tries to render the snoozed case view with active cases for a very brief moment.)

Reply:
I am going to convert `ActiveCaseList` and `SnoozedCaseList` to classes and move the logic of the `useEffect` hook into the constructor